mlx/backend/metal/make_compiled_preamble.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+OUTPUT_FILE=$1
+CC=$2
+SRCDIR=$3
+
+CONTENT=$($CC -I $SRCDIR -E $SRCDIR/mlx/backend/metal/kernels/compiled_preamble.h 2>/dev/null)
+
+cat << EOF > $OUTPUT_FILE
+// Copyright © 2023-24 Apple Inc.
```

Comment:
Lol, probably don't need this in the preamble, though technically we should have one at the top of this file.